account_asset_management/models/account_asset.py

```diff
@@ -519,12 +504,73 @@ def open_entries(self):
             "domain": [("id", "in", am_ids)],
         }
 
-    @api.multi
-    def compute_depreciation_board(self):
+    def _group_lines(self, table):
         def group_lines(x, y):
             y.update({"amount": x["amount"] + y["amount"]})
             return y
 
+        # group lines prior to depreciation start period
+        depreciation_start_date = self.date_start
+        lines = table[0]["lines"]
+        lines1 = []
+        lines2 = []
+        flag = lines[0]["date"] < depreciation_start_date
+        for line in lines:
+            if flag:
+                lines1.append(line)
+                if line["date"] >= depreciation_start_date:
+                    flag = False
+            else:
+                lines2.append(line)
+        if lines1:
+            lines1 = [reduce(group_lines, lines1)]
+            lines1[0]["depreciated_value"] = 0.0
+        table[0]["lines"] = lines1 + lines2
+
+    def _compute_depreciation_line(
+        self,
+        depreciated_value_posted,
+        table_i_start,
+        line_i_start,
+        table,
+        last_line,
+        posted_lines,
+    ):
```

Comment:
Add docstring too please.

Reply:
I think this one is also explanatory: it computes a depreciation line.